### PR TITLE
Working initial capnp based rr graph patcher.

### PR DESCRIPTION
**Pull request by litghost**
- Converts fasm2bels to use capnp for faster rr graph reading.
 - Converts xc7 routing import to use capnp for faster rr graph reading
   and writing.

### CODE REVIEW

File: utils/lib/rr_graph_capnp/graph2.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+        CAPNP_ENUM_CACHE[key] = enum_type.__dict__[camel_case_e]
+
+    return CAPNP_ENUM_CACHE[key]
+
+
+def read_switch(sw):
+    timing = sw.timing
+    sizing = sw.sizing
+
+    return graph2.Switch(
```
Comment (mithro) [inline utils/lib/rr_graph_capnp/graph2.py]: Why are the `graph2.Switch` object and the captnproto object are not compatible?

Could we make the attribute names for these objects in `graph2` just match the captnproto versions?

Comment (litghost) [reply utils/lib/rr_graph_capnp/graph2.py]: The whole point of graph2 objects is to be storage independent.  As for field names, the capnp naming is camelCase, which non of the rest of our python code is.

**Comment by litghost** [comment]
> Only one real comment -- there does seem to be a lot of conversion from `camelCaseStyle` to `camel_case_style`? Could we make things easier by just changing one side? (We can add an exception to yapf if needed too?)

The camelCase is **required** by capnp, and we just keep it isolated from the rest of our code.

File: utils/lib/rr_graph_capnp/graph2.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+            channels_obj,
+            connection_box_obj,
+            num_nodes,
+            nodes_obj,
+            num_edges,
+            edges_obj,
+            node_remap=lambda x: x
+    ):
+        """
+        Writes the routing graph to the XML file.
```
Comment (acomodi) [inline utils/lib/rr_graph_capnp/graph2.py]: This is not an XML file anymore right? I mean, it gets binarized by capnp, so maybe this comment could be misleading

Comment (litghost) [reply utils/lib/rr_graph_capnp/graph2.py]: Fixed

**Pull request merged**